tests/components/androidtv/test_media_player.py

```diff
@@ -1177,3 +1179,37 @@ async def test_connection_closed_on_ha_stop(hass):
                 hass.bus.async_fire(EVENT_HOMEASSISTANT_STOP)
                 await hass.async_block_till_done()
                 assert adb_close.called
+
+
+async def test_exception(hass):
+    """Test that the ADB connection gets closed when there is an unforeseen exception.
+
+    HA will attempt to reconnect on the next update.
+    """
+    patch_key, entity_id = _setup(CONFIG_ANDROIDTV_PYTHON_ADB)
+
+    with patchers.PATCH_ADB_DEVICE_TCP, patchers.patch_connect(True)[
+        patch_key
+    ], patchers.patch_shell(SHELL_RESPONSE_OFF)[
+        patch_key
+    ], patchers.PATCH_KEYGEN, patchers.PATCH_ANDROIDTV_OPEN, patchers.PATCH_SIGNER:
+        assert await async_setup_component(hass, DOMAIN, CONFIG_ANDROIDTV_PYTHON_ADB)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6090606)